src/mpl2/src/hier_rtlmp.cpp

```diff
@@ -3781,11 +3782,12 @@ void HierRTLMP::multiLevelMacroPlacement(Cluster* parent)
         }
         ++begin_check;
       }
-      // add early stop mechanism
-      if (best_sa) {
+      const int remaining_checks = total_runs - begin_check;
+
+      if (best_sa || remaining_checks == 0) {
```

Comment:
if remaining_checks==0 wouldn't remaining_runs ==0 too?

Reply:
I think I see your point, but the problem is the last iteration of checks.

I had a hard time writing a proper explanation, so I'll go with a practical example:

In a situation with
`Total_runs = 12`
`Threads = 8`
`Check Interval = 10`

After we end executing the first batch of 8 runs, we won't enter the check loop, because we haven't finished 10 runs yet.
..
So we execute the last 4 runs and by the time we get to the loop, `remaining_runs == 0`, so we'll check the first 10 runs and break the loop without checking the last 2. That's why we need `remaining_checks`.

Reply:
The goal is to check every 10 runs.  If we find a solution then we should ignore the remaining two runs.  Otherwise we would get a different result if we used 10 threads.

Reply:
Oh.. I'm sorry, indeed. I got confused with the checking. I'm fixing it.